### PR TITLE
Replace inline styles with CSS classes (BEM)

### PR DESCRIPTION
**Pull request by stephenburgess8**
#584 #2028 

Mastodon uses a combination of styles-in-javascript and regular css classes, mostly following the BEM naming convention.

Mostly remove javascript styles to CSS with the notable exceptions of when properties are being dynamically set in the Javascript. The goal is a pure refactor, no style changes. The biggest benefit of this would be to facilitate theme setting.

Follow a BEM naming convention. Rule of thumb: alphabetize css attributes.

Testing notes: I have compared much of the UI and have not been able to identify any regressions. However due to the large number of lines changed here, regressions seem likely. I intend to do more regression testing if the general concept is accepted by the core maintainers.

### CODE REVIEW

**Comment by nightpool** [comment]
god I wish github would let me react to posts with the 🙏  emoji

🙏 🙏 🙏

**Review by Gargron — APPROVED** [review]

File: app/assets/javascripts/components/features/status/components/detailed_status.jsx [hunk 1]
```diff
@@ -45,13 +45,13 @@ class DetailedStatus extends React.PureComponent {
...
+      applicationLink = <span> · <a className='detailed-status__application' href={status.getIn(['application', 'website'])} target='_blank' rel='nooopener'>{status.getIn(['application', 'name'])}</a></span>;
     }
 
     return (
-      <div style={{ padding: '14px 10px' }} className='detailed-status'>
-        <a href={status.getIn(['account', 'url'])} onClick={this.handleAccountClick} className='detailed-status__display-name' style={{ display: 'block', overflow: 'hidden', marginBottom: '15px' }}>
-          <div style={{ float: 'left', marginRight: '10px' }}><Avatar src={status.getIn(['account', 'avatar'])} staticSrc={status.getIn(['account', 'avatar_static'])} size={48} /></div>
+      <div className='detailed-status'>
+        <a href={status.getIn(['account', 'url'])} onClick={this.handleAccountClick} className='detailed-status__display-name'>
+          <div className='.detailed-status__display-avatar'><Avatar src={status.getIn(['account', 'avatar'])} staticSrc={status.getIn(['account', 'avatar_static'])} size={48} /></div>
```
Comment (nightpool) [inline app/assets/javascripts/components/features/status/components/detailed_status.jsx]: should be `detailed-status__display-avatar`

File: app/assets/stylesheets/components.scss [hunk 2]
```diff
@@ -611,8 +828,22 @@ a.status__content__spoiler-link {
   }
 }
 
+.detailed-status__display-avatar {
+  float: right;
```
Comment (nightpool) [inline app/assets/stylesheets/components.scss]: should be `float: left`

**Comment by stephenburgess8** [comment]
#2368

**Pull request merged**